File: app/src/main/java/com/electrosim/ConnectorInterface.java
/**
 * 
 */
package com.electrosim;

import android.graphics.Canvas;

/**
 * @author Abdul Hannan
 *
 */
public interface ConnectorInterface {

public boolean setLinkPointA (LinkPointInterface lp);
public boolean setLinkPointB (LinkPointInterface lp);
public LinkPointInterface getLinkPointA();
public LinkPointInterface getLinkPointB();
public void draw (Canvas canvas);
public void draw(Canvas canvas,float currentX, float currentY);
public void addLinkBreakpoint(LineBreakpoint point);
public boolean pointOnLine(float currentX, float currentY);
}

File: app/src/main/java/com/electrosim/ConnectorInterface.kt
/**
 * 
 */
package com.electrosim

import android.graphics.Canvas

/**
 * @author Abdul Hannan
 *
 */
interface ConnectorInterface {
    fun setLinkPointA(lp: LinkPointInterface): Boolean
    fun setLinkPointB(lp: LinkPointInterface): Boolean
    fun getLinkPointA(): LinkPointInterface
    fun getLinkPointB(): LinkPointInterface
    fun draw(canvas: Canvas)
    fun draw(canvas: Canvas, currentX: Float, currentY: Float)
    fun addLinkBreakpoint(point: LineBreakpoint)
    fun pointOnLine(currentX: Float, currentY: Float): Boolean
} 